### PR TITLE
 script: Do not do explicit reflows when handing rAFs

### PR DESCRIPTION
**Pull request by mrobinson**
This removes two explicit calls to reflow to detect rAFs that do not
modify the DOM and to trigger reflows when the page isn't dirty. This
can cause extra reflows, especially when animations are running. This
change removes them, relying on *update the rendering* to properly
trigger reflows, shortly after running rAF callbacks and after
animations are updated.

This is one part of the effort to consolidate reflows (#31871).

Signed-off-by: Martin Robinson <mrobinson@igalia.com>

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes are part of #31871.
- [x] These changes are covered by existing WPT tests. This should avoid extra frames that would normally not be detectable via the WPT harness.

### CODE REVIEW

File: components/script/dom/document.rs [hunk 1]
```diff
@@ -2145,8 +2141,8 @@ impl Document {
             // for the interim frames where we are deciding whether this rAF
             // is considered spurious, we need to ensure that the layout
             // and compositor *do* tick the animation.
-            self.window
-                .force_reflow(ReflowGoal::Full, ReflowReason::RequestAnimationFrame, None);
+            println!("Reflow will not be triggered");
```
Comment (jdm) [inline components/script/dom/document.rs:2144]: Should this be a logging call instead?

File: components/script/dom/document.rs [hunk 2]
```diff
@@ -2145,8 +2141,8 @@ impl Document {
             // for the interim frames where we are deciding whether this rAF
             // is considered spurious, we need to ensure that the layout
             // and compositor *do* tick the animation.
-            self.window
```
Comment (jdm) [inline components/script/dom/document.rs:2148]: Looks like the comment here needs updating.

File: components/script/script_thread.rs [hunk 3]
```diff
@@ -1507,13 +1507,13 @@ impl ScriptThread {
     /// <https://html.spec.whatwg.org/multipage/#update-the-rendering>
     ///
     /// Returns true if the rendering was actually updated.
-    fn update_the_rendering(&self, requested_by_compositor: bool, can_gc: CanGc) -> bool {
```
Comment (jdm) [inline components/script/script_thread.rs:1510]: Let's update the comment about the return value.

**Review by jdm — APPROVED** [review]

File: components/script/dom/document.rs (hunk 1 above)
Comment (mrobinson) [reply components/script/dom/document.rs:2144]: I had removed this `println!` locally, but neglected to commit my change. I'll remove it.

File: components/script/dom/document.rs (hunk 2 above)
Comment (mrobinson) [reply components/script/dom/document.rs:2148]: I'll update this one.

**Pull request merged**